quickwit/quickwit-search/src/root.rs

```diff
@@ -1110,7 +1109,7 @@ async fn refine_and_list_matches(
     metastore: &mut MetastoreServiceClient,
     search_request: &mut SearchRequest,
     indexes_metadata: Vec<IndexMetadata>,
-    query_ast_resolved: QueryAst,
+    mut query_ast_resolved: QueryAst,
     sort_fields_is_datetime: HashMap<String, bool>,
```

Comment:
is this really what you want here? it is a QueryAst clone that is being modified here

Reply:
Yeah, `query_ast_resolved` is last used inside this function in `list_relevant_splits`.

Reply:
it's not
```
    let tag_filter_ast = extract_tags_from_query(query_ast_resolved);
    let split_metadatas: Vec<SplitMetadata> = list_relevant_splits(
        index_uids,
        search_request.start_timestamp,
        search_request.end_timestamp,
        tag_filter_ast,
        metastore,
    )
    .await?;
```

Reply:
It is last used in a call to `extract_tags_from_query` and then only `tag_filter_ast` is used, I think I don't fully understand what you mean.